pkg/registry/generic/rest/streamer.go

```diff
@@ -56,6 +59,16 @@ func (s *LocationStreamer) InputStream(apiVersion, acceptHeader string) (stream
 	if err != nil {
 		return
 	}
+
+	if resp.StatusCode == http.StatusInternalServerError {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
```

Comment:
Sorry, I was thinking about the more generic location streamer (which can return arbitrary errors).  In this case, I'd argue that if there is error handling to be done, it's something that should be passed in as an optional interface to LocationStreamer:

    type LocationResponseChecker interface {
      Check(resp *http.Response) error
    }

Then it's the person who sets up LocationStreamer's responsibility to provide a response checker to transform the error.  LocationStreamer is really just a specific type of ResourceStreamer - an implementation class that handles the hard parts.